### PR TITLE
Add documentation for new workflow steps

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
Create Search Pipeline - https://github.com/opensearch-project/flow-framework/pull/569
Create Index - https://github.com/opensearch-project/flow-framework/pull/574
Create Ingest Pipeline - https://github.com/opensearch-project/flow-framework/pull/558

### Issues Resolved
Part of https://github.com/opensearch-project/flow-framework/issues/541

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ohltyler — APPROVED** [review]

**Review by minalsha — APPROVED** [review]

File: _automating-configurations/workflow-steps.md [hunk 1]
```diff
@@ -42,6 +42,9 @@ The following table lists the workflow step types. The `user_inputs` fields for
...
+|`create_index`|[Create Index]({{site.url}}{{site.baseurl}}/api-reference/index-apis/create-index/)     | Creates a new OpenSearch index. The inputs include `index_name`, which should be the name of the index to be created, and `configurations`, which contains the payload body of a regular REST request for creating an index.
+|`create_ingest_pipeline`|[Create Ingest Pipeline]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/) | Creates or updates an ingest pipeline. The inputs include `pipeline_id`, which should be the id of the pipeline, and `configurations`, which contains the payload body of a regular REST request for creating an ingest pipeline.
+|`create_search_pipeline`|[Create Search Pipeline]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/creating-search-pipeline/) | Creates or updates a search pipeline. The inputs include `pipeline_id`, which should be the id of the pipeline, and `configurations`, which contains the payload body of a regular REST request for creating a search pipeline.
```
Comment (dbwiddis) [inline _automating-configurations/workflow-steps.md:47]: None of the other step types in this table detail the inputs unless they have special behavior (register+deploy).

Comment (owaiskazi19) [reply _automating-configurations/workflow-steps.md:47]: These steps does have special behavior. We are passing the complete rest payload under a new field called `configurations`.

**Review by kolchfa-aws — APPROVED** [review]
LGTM. Thanks, @owaiskazi19!

File: _automating-configurations/workflow-steps.md [hunk 2]
```diff
@@ -42,6 +42,9 @@ The following table lists the workflow step types. The `user_inputs` fields for
 |`register_agent`	|[Register Agent API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/)	|Registers an agent as part of the ML Commons Agent Framework.	|
 |`delete_agent`	|[Delete Agent API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/)	|Deletes an agent.	|
 |`create_tool`	|No API	| A special-case non-API step encapsulating the specification of a tool for an agent in the ML Commons Agent Framework. These will be listed as `previous_node_inputs` for the appropriate register agent step, with the value set to `tools`.	|
+|`create_index`|[Create Index]({{site.url}}{{site.baseurl}}/api-reference/index-apis/create-index/)     | Creates a new OpenSearch index. The inputs include `index_name`, which should be the name of the index to be created, and `configurations`, which contains the payload body of a regular REST request for creating an index.
+|`create_ingest_pipeline`|[Create Ingest Pipeline]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/) | Creates or updates an ingest pipeline. The inputs include `pipeline_id`, which should be the id of the pipeline, and `configurations`, which contains the payload body of a regular REST request for creating an ingest pipeline.
```
Comment (natebower) [inline _automating-configurations/workflow-steps.md:46]:
```suggestion
|`create_ingest_pipeline`|[Create Ingest Pipeline]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/) | Creates or updates an ingest pipeline. The inputs include `pipeline_id`, which should be the ID of the pipeline, and `configurations`, which contains the payload body of a regular REST request for creating an ingest pipeline.
```

File: _automating-configurations/workflow-steps.md (hunk 1 above)
Comment (natebower) [inline _automating-configurations/workflow-steps.md:47]:
```suggestion
|`create_search_pipeline`|[Create Search Pipeline]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/creating-search-pipeline/) | Creates or updates a search pipeline. The inputs include `pipeline_id`, which should be the ID of the pipeline, and `configurations`, which contains the payload body of a regular REST request for creating a search pipeline.
```

**Review by natebower — APPROVED** [review]
@owaiskazi19 @kolchfa-aws Just two minor changes. Thanks!

**Pull request merged**